### PR TITLE
[nextercism] Detect the type of a path

### PR DESCRIPTION
**Pull request by kytrinyx**
We are pretty flexible in the type of argument that we receive. It might be
an actual path, or it might be a conceptual path: the name of an exercise that
we need to walk the workspace to find. If it's a conceptual path we might find
more than one, in which case we'll do an interactive thing to disambiguate. If
it's a real path, then we can do a lot less guessing.

This is in preparation for the submit command.
We need to figure out what the argument to the submit command is. Or if multiple, what each of them are.

### CODE REVIEW

File: workspace/path_type.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+// DetectPathType determines whether the given path is a directory, a file, or the name of an exercise.
+func DetectPathType(path string) (PathType, error) {
+	// If it's not an absolute path, make it one.
+	if !filepath.IsAbs(path) {
+		var err error
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return 0, err
```
Comment (nywilken) [inline workspace/path_type.go]: This is a little confusing seeing as TypeExerciseID is also 0. What do you think about starting the constants at 1 or maybe returning a nagative number.

Comment (Tonkpils) [inline workspace/path_type.go]: > returning a nagative number

👍

Comment (kytrinyx) [inline workspace/path_type.go]: I like the idea of returning a negative number. Very clean!

**Comment by kytrinyx** [comment]
This should be good to go now.

**Pull request merged**